bindings/matrix-sdk-ffi/src/sliding_sync.rs

```diff
@@ -775,7 +775,10 @@ impl Client {
 impl Client {
     pub fn sliding_sync(&self) -> Arc<SlidingSyncBuilder> {
         RUNTIME.block_on(async move {
-            let inner = self.client.sliding_sync().await;
+            let mut inner = self.client.sliding_sync().await;
+            if let Some(sliding_sync_proxy) = self.client.sliding_sync_proxy().await {
+                inner = inner.homeserver(sliding_sync_proxy);
+            }
```

Comment:
This is untested, but as I understand it should allow sliding sync to use the discovered server, but also for a client to override this by calling `.homeserver(myServerURLString)` if desired.